src/DependencyInjection/Configuration.php

```diff
@@ -31,6 +32,11 @@ public function getConfigTreeBuilder(): TreeBuilder
                     ->defaultValue('ROLE_OAUTH2_')
                     ->cannotBeEmpty()
                 ->end()
+                ->scalarNode('anonymous_user_identifier')
```

Comment:
Can you please remind me under which flow do we expect `anonymous` users? Wording-wise the `anonymous` concept has been removed in Symfony but I feel like this one is different and specific to this bundle, what's the case I'm missing?

Reply:
The `AuthorizationController` does not require authentication (although it is highly recommended), so when the `AuthorizationRequestResolveEvent` is dispatched, there is a valid use case where the `AuthorizationRequestUserResolvingListener` cannot set the user instance in the event:
https://github.com/thephpleague/oauth2-server-bundle/blob/ec2c2d3a13f1f4b02de4c0472f0b5bc3bd1db33a/src/EventListener/AuthorizationRequestUserResolvingListenerTrait.php#L12-L18

So the `$event->getUser()` call can return null with a resolved authorization https://github.com/thephpleague/oauth2-server-bundle/blob/ec2c2d3a13f1f4b02de4c0472f0b5bc3bd1db33a/src/Controller/AuthorizationController.php#L104

This is the use case I think we should disallow.